Ignore special argument names on Parameter Documentation checker

<!--
  Hi there! Thank you for wanting to make pylint better.

  Before you submit this, make sure that this feature wasn't
  already requested or if it is not already implemented in the master branch.
-->

### Is your feature request related to a problem? Please describe
Yes. Pylint handles special variables names (`_`, `__`, `ignored_`, etc) very well in most cases. However, it doesn't have the same behaviour for *Parameter Documentation checker*.

This code:

```python 
def function_foo(name, _):
    """function foo ...

    Parameters
    ----------
    name: str
        description
    """
    print(name)
```

Will complain about not documenting the parameter "_"

```
************* Module pylint_config
pylint_config.py:1:0: W9015: "_" missing in parameter documentation (missing-param-doc)
pylint_config.py:1:0: W9016: "_" missing in parameter type documentation (missing-type-doc)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Describe the solution you'd like
It would be helpful if the [ignored argument names](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/variables.py#L67-L68) were also ignored on docstrings.

Alternatively, a way for disabling the check for ignored arguments would also work. 

Or even a regex config to be used on `.pylintrc` like:

```
missing-param-doc=^(_) # ignore docs for parameters starting with _
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -403,3 +403,5 @@ contributors:
 * Yeting Li (yetingli): contributor
 
 * Frost Ming (frostming): contributor
+
+* Luigi Bertaco Cristofolini (luigibertaco): contributor

--- a/ChangeLog
+++ b/ChangeLog
@@ -23,7 +23,7 @@ Release date: TBA
 
 * Fix a crash when a specified config file does not exist
 
-* Add support to ``ignored-argument-names`` in DocstringParameterChecker.
+* Add support to ``ignored-argument-names`` in DocstringParameterChecker and adds `useless-param-doc` and `useless-type-doc` messages.
 
   Close #3800
 

--- a/ChangeLog
+++ b/ChangeLog
@@ -23,6 +23,10 @@ Release date: TBA
 
 * Fix a crash when a specified config file does not exist
 
+* Add support to ``ignored-argument-names`` in DocstringParameterChecker.
+
+  Close #3800
+
 What's New in Pylint 2.6.0?
 ===========================
 

--- a/doc/whatsnew/2.7.rst
+++ b/doc/whatsnew/2.7.rst
@@ -9,6 +9,8 @@
 Summary -- Release highlights
 =============================
 
+* Add support to ``ignored-argument-names`` in DocstringParameterChecker and
+  adds `useless-param-doc` and `useless-type-doc` messages.
 
 New checkers
 ============

--- a/pylint/extensions/docparams.py
+++ b/pylint/extensions/docparams.py
@@ -125,6 +125,16 @@ class DocstringParameterChecker(BaseChecker):
             "differing-type-doc",
             "Please check parameter names in type declarations.",
         ),
+        "W9019": (
+            '"%s" useless ignored parameter documentation',
+            "useless-param-doc",
+            "Please check ignored parameter names in declarations.",
+        ),
+        "W9020": (
+            '"%s" useless ignored parameter type documentation',
+            "useless-type-doc",
+            "Please check ignored parameter names in type declarations.",
+        ),
     }
 
     options = (
@@ -444,11 +454,10 @@ class DocstringParameterChecker(BaseChecker):
         expected_argument_names.update(arg.name for arg in arguments_node.kwonlyargs)
         not_needed_type_in_docstring = self.not_needed_param_in_docstring.copy()
 
-        ignored_argument_names = get_global_option(
-            self, "ignored-argument-names", set()
-        )
+        expected_but_ignored_argument_names = set()
+        ignored_argument_names = get_global_option(self, "ignored-argument-names")
         if ignored_argument_names:
-            ignored_argument_names = {
+            expected_but_ignored_argument_names = {
                 arg
                 for arg in expected_argument_names
                 if ignored_argument_names.match(arg)
@@ -471,7 +480,7 @@ class DocstringParameterChecker(BaseChecker):
                 params_with_doc,
                 "missing-param-doc",
                 self.not_needed_param_in_docstring,
-                expected_argument_names - ignored_argument_names,
+                expected_argument_names - expected_but_ignored_argument_names,
                 warning_node,
             )
 
@@ -487,7 +496,7 @@ class DocstringParameterChecker(BaseChecker):
                 params_with_type,
                 "missing-type-doc",
                 not_needed_type_in_docstring,
-                expected_argument_names - ignored_argument_names,
+                expected_argument_names - expected_but_ignored_argument_names,
                 warning_node,
             )
 
@@ -503,6 +512,20 @@ class DocstringParameterChecker(BaseChecker):
             "differing-type-doc",
             not_needed_type_in_docstring,
             expected_argument_names,
+            warning_node,
+        )
+        self._compare_different_args(
+            params_with_doc,
+            "useless-param-doc",
+            self.not_needed_param_in_docstring,
+            expected_argument_names - expected_but_ignored_argument_names,
+            warning_node,
+        )
+        self._compare_different_args(
+            params_with_type,
+            "useless-type-doc",
+            not_needed_type_in_docstring,
+            expected_argument_names - expected_but_ignored_argument_names,
             warning_node,
         )
 

--- a/pylint/extensions/docparams.py
+++ b/pylint/extensions/docparams.py
@@ -348,16 +348,20 @@ class DocstringParameterChecker(BaseChecker):
         """Compare the found argument names with the expected ones and
         generate a message if there are arguments missing.
 
-        :param set found_argument_names: argument names found in the
-            docstring
+        :param found_argument_names: argument names found in the docstring
+        :type found_argument_names: set
 
-        :param str message_id: pylint message id
+        :param message_id: pylint message id
+        :type message_id: str
 
         :param not_needed_names: names that may be omitted
-        :type not_needed_names: set of str
+        :type not_needed_names: set
 
-        :param set expected_argument_names: Expected argument names
-        :param NodeNG warning_node: The node to be analyzed
+        :param expected_argument_names: Expected argument names
+        :type expected_argument_names: set
+
+        :param warning_node: The node to be analyzed
+        :type warning_node: :class:`astroid.scoped_nodes.Node`
         """
         missing_argument_names = (
             expected_argument_names - found_argument_names
@@ -380,16 +384,20 @@ class DocstringParameterChecker(BaseChecker):
         """Compare the found argument names with the expected ones and
         generate a message if there are extra arguments found.
 
-        :param set found_argument_names: argument names found in the
-            docstring
+        :param found_argument_names: argument names found in the docstring
+        :type found_argument_names: set
 
-        :param str message_id: pylint message id
+        :param message_id: pylint message id
+        :type message_id: str
 
         :param not_needed_names: names that may be omitted
-        :type not_needed_names: set of str
+        :type not_needed_names: set
 
-        :param set expected_argument_names: Expected argument names
-        :param NodeNG warning_node: The node to be analyzed
+        :param expected_argument_names: Expected argument names
+        :type expected_argument_names: set
+
+        :param warning_node: The node to be analyzed
+        :type warning_node: :class:`astroid.scoped_nodes.Node`
         """
         differing_argument_names = (
             (expected_argument_names ^ found_argument_names)
@@ -414,13 +422,17 @@ class DocstringParameterChecker(BaseChecker):
         """Compare the found argument names with the ignored ones and
         generate a message if there are ignored arguments found.
 
-        :param set found_argument_names: argument names found in the
-            docstring
+        :param found_argument_names: argument names found in the docstring
+        :type found_argument_names: set
 
-        :param str message_id: pylint message id
+        :param message_id: pylint message id
+        :type message_id: str
 
-        :param set ignored_argument_names: Ignored argument names
-        :param NodeNG warning_node: The node to be analyzed
+        :param expected_argument_names: Expected argument names
+        :type expected_argument_names: set
+
+        :param warning_node: The node to be analyzed
+        :type warning_node: :class:`astroid.scoped_nodes.Node`
         """
         existing_ignored_argument_names = ignored_argument_names & found_argument_names
 

--- a/pylint/extensions/docparams.py
+++ b/pylint/extensions/docparams.py
@@ -26,6 +26,7 @@ from pylint.checkers import BaseChecker
 from pylint.checkers import utils as checker_utils
 from pylint.extensions import _check_docs_utils as utils
 from pylint.interfaces import IAstroidChecker
+from pylint.utils import get_global_option
 
 
 class DocstringParameterChecker(BaseChecker):
@@ -443,6 +444,16 @@ class DocstringParameterChecker(BaseChecker):
         expected_argument_names.update(arg.name for arg in arguments_node.kwonlyargs)
         not_needed_type_in_docstring = self.not_needed_param_in_docstring.copy()
 
+        ignored_argument_names = get_global_option(
+            self, "ignored-argument-names", set()
+        )
+        if ignored_argument_names:
+            ignored_argument_names = {
+                arg
+                for arg in expected_argument_names
+                if ignored_argument_names.match(arg)
+            }
+
         if arguments_node.vararg is not None:
             expected_argument_names.add(arguments_node.vararg)
             not_needed_type_in_docstring.add(arguments_node.vararg)
@@ -460,7 +471,7 @@ class DocstringParameterChecker(BaseChecker):
                 params_with_doc,
                 "missing-param-doc",
                 self.not_needed_param_in_docstring,
-                expected_argument_names,
+                expected_argument_names - ignored_argument_names,
                 warning_node,
             )
 
@@ -476,7 +487,7 @@ class DocstringParameterChecker(BaseChecker):
                 params_with_type,
                 "missing-type-doc",
                 not_needed_type_in_docstring,
-                expected_argument_names,
+                expected_argument_names - ignored_argument_names,
                 warning_node,
             )
 

--- a/pylint/extensions/docparams.py
+++ b/pylint/extensions/docparams.py
@@ -428,8 +428,8 @@ class DocstringParameterChecker(BaseChecker):
         :param message_id: pylint message id
         :type message_id: str
 
-        :param expected_argument_names: Expected argument names
-        :type expected_argument_names: set
+        :param ignored_argument_names: Expected argument names
+        :type ignored_argument_names: set
 
         :param warning_node: The node to be analyzed
         :type warning_node: :class:`astroid.scoped_nodes.Node`

--- a/pylint/extensions/docparams.py
+++ b/pylint/extensions/docparams.py
@@ -128,12 +128,12 @@ class DocstringParameterChecker(BaseChecker):
         "W9019": (
             '"%s" useless ignored parameter documentation',
             "useless-param-doc",
-            "Please check ignored parameter names in declarations.",
+            "Please remove the ignored parameter documentation.",
         ),
         "W9020": (
             '"%s" useless ignored parameter type documentation',
             "useless-type-doc",
-            "Please check ignored parameter names in type declarations.",
+            "Please remove the ignored parameter type documentation.",
         ),
     }
 

--- a/tests/extensions/test_check_docs.py
+++ b/tests/extensions/test_check_docs.py
@@ -2170,7 +2170,7 @@ class TestParamDocChecker(CheckerTestCase):
                 :param arg: An argument.
                 :type arg: int
                 '''
-                raise NotImplementedError()
+                pass
         """
         )
         with self.assertNoMessages():
@@ -2189,7 +2189,7 @@ class TestParamDocChecker(CheckerTestCase):
                 Args:
                     arg (int): An argument.
                 '''
-                raise NotImplementedError()
+                pass
         """
         )
         with self.assertNoMessages():
@@ -2210,20 +2210,20 @@ class TestParamDocChecker(CheckerTestCase):
                 arg : int
                     An argument.
                 '''
-                raise NotImplementedError()
+                pass
         """
         )
         with self.assertNoMessages():
             self.checker.visit_functiondef(node)
 
-    def test_accepts_ignored_argument_names_sphinx(self):
+    def test_useless_docs_ignored_argument_names_sphinx(self):
         """Example of a method documenting the return type that an
         implementation should return.
         """
         node = astroid.extract_node(
             """
         class Foo(object):
-            def foo(self, arg, _): #@
+            def foo(self, arg, _, _ignored): #@
                 '''docstring ...
 
                 :param arg: An argument.
@@ -2231,41 +2231,50 @@ class TestParamDocChecker(CheckerTestCase):
 
                 :param _: Another argument.
                 :type _: float
+
+                :param _ignored: Ignored argument.
                 '''
-                raise NotImplementedError()
+                pass
         """
         )
-        with self.assertNoMessages():
+        with self.assertAddsMessages(
+            Message(msg_id="useless-param-doc", node=node, args=("_, _ignored",)),
+            Message(msg_id="useless-type-doc", node=node, args=("_",)),
+        ):
             self.checker.visit_functiondef(node)
 
-    def test_accepts_ignored_argument_names_google(self):
+    def test_useless_docs_ignored_argument_names_google(self):
         """Example of a method documenting the return type that an
         implementation should return.
         """
         node = astroid.extract_node(
             """
         class Foo(object):
-            def foo(self, arg, _): #@
+            def foo(self, arg, _, _ignored): #@
                 '''docstring ...
 
                 Args:
                     arg (int): An argument.
                     _ (float): Another argument.
+                    _ignored: Ignored argument.
                 '''
-                raise NotImplementedError()
+                pass
         """
         )
-        with self.assertNoMessages():
+        with self.assertAddsMessages(
+            Message(msg_id="useless-param-doc", node=node, args=("_, _ignored",)),
+            Message(msg_id="useless-type-doc", node=node, args=("_",)),
+        ):
             self.checker.visit_functiondef(node)
 
-    def test_accepts_ignored_argument_names_numpy(self):
+    def test_useless_docs_ignored_argument_names_numpy(self):
         """Example of a method documenting the return type that an
         implementation should return.
         """
         node = astroid.extract_node(
             """
         class Foo(object):
-            def foo(self, arg, _): #@
+            def foo(self, arg, _, _ignored): #@
                 '''docstring ...
 
                 Parameters
@@ -2275,9 +2284,15 @@ class TestParamDocChecker(CheckerTestCase):
 
                 _ : float
                     Another argument.
+
+                _ignored :
+                    Ignored Argument
                 '''
-                raise NotImplementedError()
+                pass
         """
         )
-        with self.assertNoMessages():
+        with self.assertAddsMessages(
+            Message(msg_id="useless-param-doc", node=node, args=("_, _ignored",)),
+            Message(msg_id="useless-type-doc", node=node, args=("_",)),
+        ):
             self.checker.visit_functiondef(node)

--- a/tests/extensions/test_check_docs.py
+++ b/tests/extensions/test_check_docs.py
@@ -2156,3 +2156,128 @@ class TestParamDocChecker(CheckerTestCase):
         )
         with self.assertNoMessages():
             self.checker.visit_functiondef(node)
+
+    def test_ignores_ignored_argument_names_sphinx(self):
+        """Example of a method documenting the return type that an
+        implementation should return.
+        """
+        node = astroid.extract_node(
+            """
+        class Foo(object):
+            def foo(self, arg, _): #@
+                '''docstring ...
+
+                :param arg: An argument.
+                :type arg: int
+                '''
+                raise NotImplementedError()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_ignores_ignored_argument_names_google(self):
+        """Example of a method documenting the return type that an
+        implementation should return.
+        """
+        node = astroid.extract_node(
+            """
+        class Foo(object):
+            def foo(self, arg, _): #@
+                '''docstring ...
+
+                Args:
+                    arg (int): An argument.
+                '''
+                raise NotImplementedError()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_ignores_ignored_argument_names_numpy(self):
+        """Example of a method documenting the return type that an
+        implementation should return.
+        """
+        node = astroid.extract_node(
+            """
+        class Foo(object):
+            def foo(self, arg, _): #@
+                '''docstring ...
+
+                Parameters
+                ----------
+                arg : int
+                    An argument.
+                '''
+                raise NotImplementedError()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_accepts_ignored_argument_names_sphinx(self):
+        """Example of a method documenting the return type that an
+        implementation should return.
+        """
+        node = astroid.extract_node(
+            """
+        class Foo(object):
+            def foo(self, arg, _): #@
+                '''docstring ...
+
+                :param arg: An argument.
+                :type arg: int
+
+                :param _: Another argument.
+                :type _: float
+                '''
+                raise NotImplementedError()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_accepts_ignored_argument_names_google(self):
+        """Example of a method documenting the return type that an
+        implementation should return.
+        """
+        node = astroid.extract_node(
+            """
+        class Foo(object):
+            def foo(self, arg, _): #@
+                '''docstring ...
+
+                Args:
+                    arg (int): An argument.
+                    _ (float): Another argument.
+                '''
+                raise NotImplementedError()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
+    def test_accepts_ignored_argument_names_numpy(self):
+        """Example of a method documenting the return type that an
+        implementation should return.
+        """
+        node = astroid.extract_node(
+            """
+        class Foo(object):
+            def foo(self, arg, _): #@
+                '''docstring ...
+
+                Parameters
+                ----------
+                arg : int
+                    An argument.
+
+                _ : float
+                    Another argument.
+                '''
+                raise NotImplementedError()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)

